ArduCopter/esc_calibration.cpp

```diff
@@ -151,30 +160,27 @@ void Copter::esc_calibration_auto()
     // disable safety if requested
     BoardConfig.init_safety();
 
-    // arm and enable motors
-    motors->armed(true);
-    SRV_Channels::enable_by_mask(motors->get_motor_mask());
-    hal.util->set_soft_armed(true);
-
-    // flash LEDs
-    esc_calibration_notify();
-
-    // raise throttle to maximum
-    delay(10);
-
     // wait for safety switch to be pressed
     while (hal.util->safety_switch_state() == AP_HAL::Util::SAFETY_DISARMED) {
+        bool printed_msg = false;
         if (!printed_msg) {
             gcs().send_text(MAV_SEVERITY_INFO,"ESC calibration: Push safety switch");
             printed_msg = true;
         }
-        SRV_Channels::cork();
-        motors->set_throttle_passthrough_for_esc_calibration(1.0f);
-        SRV_Channels::push();
         esc_calibration_notify();
         delay(3);
     }
 
+    // arm and enable motors
+    motors->armed(true);
+    SRV_Channels::enable_by_mask(motors->get_motor_mask());
+    hal.util->set_soft_armed(true);
+
+    // raise throttle to maximum
+    SRV_Channels::cork();
+    motors->set_throttle_passthrough_for_esc_calibration(1.0f);
+    SRV_Channels::push();
+
     // delay for 5 seconds while outputting pulses
     uint32_t tstart = millis();
     while (millis() - tstart < 5000) {
```

Comment:
I hate that GitHub doesn't allow me to comment on the line I want!

Unrelated to this PR:
- line below (and line (old) 191/(new) 197): is there any reason why we need it?
- line (old) 187/(new) 193 should probably be surrounded by cork and push

Reply:
I agree there should be no need for the duplicate, "motors->set_throttle_passthrough_for_esc_calibration(0.0f);". i'll remove that.